iOS-TvShows/iOS-TvShows/Modules/Shows/ShowsPresenter.swift

```diff
@@ -41,15 +41,29 @@ extension ShowsPresenter: ShowsPresenterInterface {
 
     func configure(with output: Shows.ViewOutput) -> Shows.ViewInput {
         handle(settings: output.settings)
-        return Shows.ViewInput()
+        return Shows.ViewInput(
+            shows: createItems()
+        )
     }
 
     func handle(settings: Signal<Void>) {
         settings
-            .emit(onNext: { [unowned self] _ in
+            .emit(onNext: { [unowned self] in
                 wireframe.goToSettings()
             })
             .disposed(by: disposeBag)
     }
 
+    func createItems() -> Driver<[TableCellItem]> {
+        interactor
+            .shows
+            .handleLoadingAndError(with: view)
+            .asDriver(onErrorDriveWith: .never())
+            .map { [unowned self] in $0.map {
```

Comment:
We are avoiding doing lots of logic inside closure in Rx chains, rather move this mapping to separate method, something like this (also better name would be fetchShows since you are fetching it, creating is just a small part of this method time-wise):

```swift
extension ShowsPresenter: ShowsPresenterInterface {

    func configure(with output: Shows.ViewOutput) -> Shows.ViewInput {
        handle(settings: output.settings)
        let shows = fetchShows()
        let items = shows
            .map { [unowned self] in createItems(from: $0) }

        return Shows.ViewInput(
            shows: items.map { $0 as [TableCellItem] } // Swift compiler issue
        )
    }

    func handle(settings: Signal<Void>) {
        settings
            .emit(onNext: { [unowned self] in
                wireframe.goToSettings()
            })
            .disposed(by: disposeBag)
    }

    func fetchShows() -> Driver<[Show]> {
        return interactor
            .shows
            .handleLoadingAndError(with: view)
            .asDriver(onErrorJustReturn: [])
    }

    func createItems(from shows: [Show]) -> [ShowTableCellItem] {
        return shows.map {
            return ShowTableCellItem(
                show: $0,
                didSelect: { [unowned wireframe] in wireframe.goToShowDetails(with: $0.show) }
            )
        }
    }
}

```